lib/exitCodes.ts

```diff
@@ -1,28 +1,79 @@
 import {Logger} from './logger2';
 
 const CONFIG_ERROR_CODE = 105;
+const BROWSER_ERROR_CODE = 135;
+
+export class ProtractorError extends Error {
+  static ERR_MSGS: string[];
+  static DEFAULT_MSG = 'protractor encountered an error';
 
-export class ProtractorError {
   error: Error;
   description: string;
   code: number;
   stack: string;
   constructor(logger: Logger, description: string, code: number) {
+    super();
     this.error = new Error();
```

Comment:
This was a weird work around because `stack` is not being part of `node` typings. Using `captureStackTrace` with `any` to get the stack.